connection_manager.go

```diff
@@ -206,6 +256,10 @@ func (cm *connectionManager) createConnection() (*connection, error) {
 }
 
 func (cm *connectionManager) get() (*connection, error) {
+	if cm.connectionCounter.isLessThan(cm.minConnections) {
+		return cm.create()
+	}
```

Comment:
I think that could work, the only potential problem I could see is that if there are temporary issues getting a new connection, but there are existing connections available to use then this would starve any commands from executing.

Thoughts?